### ISSUE TITLE
False positive for IDE0051 

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn#32727]
**Version Used**: VS2019 Preview 2

**Steps to Reproduce**:

Minimal code to reproduce:

````csharp
using System.Runtime.InteropServices;

class Program
{
    private const int MAX_PATH = 260; // IDE0051 unused field

    [StructLayout(LayoutKind.Sequential)]
    internal struct ProcessEntry32
    {
        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = MAX_PATH)]
        public string szExeFile;
    }
}
````

**Expected Behavior**:

`MAX_PATH` should not be reported as unused as there is a usage in the attribute `MarshalAs`.

**Actual Behavior**:

`MAX_PATH` is reported as unused.

**Comment by mavasani** [comment]
Interesting, given we have a test for a similar scenario: http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs,734. I will investigate.

**Comment by meziantou** [comment]
BTW, it looks like the attribute `[StructLayout(LayoutKind.Sequential)]` is important in the repro. Removing the attribute also removes the diagnostic.

**Comment by mavasani** [comment]
Ah, I think I now get it - we bail out completely from analysis of types marked with StructLayout [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs,157). Instead we should continue with the analysis and just bail out reporting unused fields/properties for such a type. I will have a PR out soon. Thanks @meziantou!

**Comment by rdeago** [comment]
The bug is still present, just as described in the original post, in **Visual Studio 2022 v17.5.5**.

Here's an extract of version information from the About dialog:

```text
Microsoft Visual Studio Community 2022
Version 17.5.5
VisualStudio.17.Release/17.5.5+33627.172

C# Tools   4.5.2-3.23171.7+d17f741546fad2786cbd6394d08619544e53a36d
```

**Comment by mavasani** [comment]
> The bug seems to be still present in **Visual Studio 2022 v17.5.5**.
> 
> No struct layout involved this time, so maybe it's not _exactly_ the same bug... 🤔
> 
> Happens in an ASP.NET application using Minimal APIs, compiling for .NET 7.
> 
> ```cs
> // IDE0051: Private member 'MyHeaderName' is unused
> private const string MyHeaderName = "X-My-Header";
> 
> // ...
> 
> app.MapGet("/some/path", static ([FromHeader(Name = MyHeaderName)] string myHeader) => Results.Ok);
> ```
> 
> Apparently, a constant used _only_ as an attribute parameter is not... used enough, or something.
> 
> Not a show stopper, but annoying.

Can you please file a separate issue for this? Thanks

**Comment by rdeago** [comment]
> Can you please file a separate issue for this? Thanks

Of course. Just filed #68162.
